operator_ui/src/pages/Jobs/New.tsx

```diff
@@ -0,0 +1,270 @@
+    const definition: OcrJobSpecRequest = { toml: value }
+    return api.v2.ocrSpecs.createJobSpec(definition)
+  }
+
+  return Promise.reject('Invalid format')
+}
+
+function getInitialValues({
+  query,
+}: {
+  query: string
+}): { jobSpec: string; format: JobSpecFormats } {
+  const params = new URLSearchParams(query)
+  const jobSpec = (params.get('definition') as string) || getPersistJobSpec()
+
+  const format =
+    getJobSpecFormat({
+      value: jobSpec,
+    }) ||
+    JobSpecFormats[params.get('format')?.toUpperCase() as JobSpecFormat] ||
+    (storage.get(SELECTED_FORMAT) as JobSpecFormat) ||
+    JobSpecFormats.JSON
+
+  return {
+    jobSpec: stringifyJobSpec({ value: jobSpec || '' }),
+    format: format || JobSpecFormats.JSON,
+  }
+}
+
+export const New = ({
+  classes,
+}: {
+  classes: WithStyles<typeof styles>['classes']
+}) => {
+  const location = useLocation()
+  const [initialValues] = React.useState(
+    getInitialValues({
+      query: location.search,
+    }),
+  )
+
+  const [format, setFormat] = React.useState<JobSpecFormats>(
+    initialValues.format,
+  )
+  const [value, setValue] = React.useState<string>(initialValues.jobSpec)
+  const [valid, setValid] = React.useState<boolean>(true)
+  const [loading, setLoading] = React.useState<boolean>(false)
+  const dispatch = useDispatch()
+  const history = useHistory()
+
+  React.useEffect(() => {
+    setPersistJobSpec(value)
+    setValid(true)
+  }, [value])
+
+  React.useEffect(() => {
+    storage.set(SELECTED_FORMAT, format)
+    setValid(true)
+    history.replace({
+      search: `?format=${format}`,
+    })
+  }, [format, history])
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+    const isValid = validate({ format, value })
+    setValid(isValid)
+
+    if (isValid) {
+      setLoading(true)
+
+      apiCall({
+        format,
+        value,
+      })
+        .then(({ data }) => {
+          dispatch(notifySuccess(SuccessNotification, data))
+        })
+        .catch((error) => {
+          dispatch(notifyError(ErrorMessage, error))
+          setValid(false)
+        })
+        .finally(() => {
+          setLoading(false)
+        })
+    }
+  }
+
+  return (
+    <Content>
+      <Grid container spacing={40}>
+        <Grid item xs={12} md={11} lg={9}>
+          <Card className={classes.card}>
+            <CardHeader title="New Job" />
+            <Divider />
+            <CardContent>
+              <form noValidate onSubmit={handleSubmit}>
+                <Grid container>
+                  <Grid item xs={12}>
+                    <FormControl fullWidth>
+                      <FormLabel>Job Spec Format</FormLabel>
+                      <RadioGroup
+                        name="select-format"
+                        value={format}
+                        onChange={(event: any) =>
+                          setFormat(event.target.value as JobSpecFormats)
```

Comment:
Should we clean the value too? Right now if I change the format and have JSON job spec inside it stays there. Might be nice to have separate storage for both formats too.

Reply:
I was thinking about it. But i came to conclusion it is not the best idea. It adds some complex to the code and from UX perspective your are still on one form.